CMakeLists.txt

```diff
@@ -123,6 +123,18 @@ else ()
     message (AUTHOR_WARNING "${ISSUE_REPORT_MSG}")
 endif ()
 
+# MSVC symbol visibility and linker behavior is sufficiently different from all
+# other platforms. Here, we ensure tests/tools built as EXE choose symbol
+# visibility appropriate for linking the library, whether static or dynamic.
+if (MSVC)
```

Comment:
What happens with MINGW or other compilers (Borland?) on Win32 platforms?  Put another way, is the issue here the *compiler* (MSVC) or the *platform* (Windows)?  I don't know the answer, but I hope you or someone else here might.